### PR TITLE
UCT/CUDA-IPC: strncmp returns 0 on match but khash expects 1 on match

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
- showed up when khash was tested and there was 100% cache miss

@bureddy @yosefe @brminich

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: src/uct/cuda/cuda_ipc/cuda_ipc_ep.h [hunk 1]
```diff
@@ -33,8 +33,8 @@ static inline khint_t uct_cuda_ipc_memh_hash_func(CUipcMemHandle seg)
     return hash_val;
 }
 
-#define uct_cuda_ipc_memh_hash_equal(_sg1, _sg2)  \
-    strncmp((const char *) &(_sg1), (const char *) &(_sg2), sizeof(CUipcMemHandle))
+#define uct_cuda_ipc_memh_hash_equal(_sg1, _sg2)                        \
```
Comment (brminich) [inline src/uct/cuda/cuda_ipc/cuda_ipc_ep.h]: why need these spaces?

Comment (Akshay-Venkatesh) [reply src/uct/cuda/cuda_ipc/cuda_ipc_ep.h]: This was because of emacs region indentation. These change was automatically applied for the macro.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7151/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@Akshay-Venkatesh can you pls address the https://github.com/openucx/ucx/pull/2652#discussion_r191721032 in another PR?
@bureddy i guess we need this in v1.4.x branch as well..

File: src/uct/cuda/cuda_ipc/cuda_ipc_ep.h (hunk 1 above)
Comment (Akshay-Venkatesh) [reply src/uct/cuda/cuda_ipc/cuda_ipc_ep.h]: @brminich @yosefe 
Do I need to remove the spaces even if was automatically done by emacs?

**Pull request merged**